Candy87 PCB Firmware

rarepotato8de opened the pull request:
Candy87 Firmware

## Description

This is the initial firmware for the Candy87 PCB that is offered by Candykeys. 
This PCB was designed for the Geonworks F1-8X V2.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

rarepotato8de committed:
candy87 init

candy87 v2 firmware, working and tested

zvecr commented on keyboards/candykeys/candy87/rules.mk:
This file should be removed

zvecr commented on keyboards/candykeys/candy87/readme.md:
```diff
@@ -0,0 +1,29 @@
+![Candy87](https://i.imgur.com/HeCz1u8.png)
+
+The Candy87 Hotswap PCB is a universal TKL PCB by Candykeys.
+It supports ISO Hotswap by default (ANSI possible with desolering two hotswap sockets) and many layout options. 
```
> and many layout options. 

So why is only a single layout defined within the json config?

rarepotato8de replied:
This is one "ALL" layout, so it has all of the different layouts and keys in this one layout.
On some keyboards these layouts were named "LAYOUT_ALL". It just has so many combination of layouts that this "ALL" layout is more convenient

zvecr replied:
I think you are misunderstanding the question, and trying to explain something that I, a maintainer of QMK, are already aware of.

Having a single "all" layout macro is invalid. It not being called `LAYOUT_all` to avoid the lint warning is also invalid

You need to add more layouts.

rarepotato8de replied:
Please apologize. I added many (community) layouts to it now

rarepotato8de replied:
Added keymaps for all layouts - should be fine now

zvecr commented on keyboards/candykeys/candy87/info.json:
This file needs to be renamed to `keyboard.json`

rarepotato8de replied:
done

zvecr commented on keyboards/candykeys/candy87/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,51 @@
+  KC_TRNS,  KC_TRNS,  KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,       KC_TRNS,      KC_TRNS,      KC_TRNS,      KC_TRNS,    KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,       		 KC_TRNS ,   KC_TRNS,    KC_TRNS,
+  KC_TRNS,  KC_TRNS,  KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,       KC_TRNS,      KC_TRNS,      KC_TRNS,      KC_TRNS,    KC_TRNS,   KC_TRNS, KC_TRNS,  
+  KC_TRNS,  KC_TRNS,  KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,       KC_TRNS,      KC_TRNS,      KC_TRNS,      KC_TRNS,    KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,            		         KC_TRNS,
+  KC_TRNS,  KC_TRNS,  KC_TRNS,                               KC_TRNS,                                                             KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,              KC_TRNS,    KC_TRNS,    KC_TRNS),
+
+[2] = LAYOUT(
+  KC_TRNS,  KC_TRNS,  KC_TRNS,     KC_TRNS,    KC_TRNS,      KC_TRNS,       KC_TRNS,      KC_TRNS,      KC_TRNS,      KC_TRNS,    KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,      		 KC_TRNS,    KC_TRNS,    KC_TRNS,
+  KC_TRNS,  KC_TRNS,  KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,       KC_TRNS,      KC_TRNS,      KC_TRNS,      KC_TRNS,    KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,     KC_TRNS,    KC_TRNS,    KC_TRNS,
+  KC_TRNS,  KC_TRNS,  KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,       KC_TRNS,      KC_TRNS,      KC_TRNS,      KC_TRNS,    KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,       		 KC_TRNS ,   KC_TRNS,    KC_TRNS,
+  KC_TRNS,  KC_TRNS,  KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,       KC_TRNS,      KC_TRNS,      KC_TRNS,      KC_TRNS,    KC_TRNS,   KC_TRNS, KC_TRNS,  
+  KC_TRNS,  KC_TRNS,  KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,       KC_TRNS,      KC_TRNS,      KC_TRNS,      KC_TRNS,    KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,            		         KC_TRNS,
+  KC_TRNS,  KC_TRNS,  KC_TRNS,                               KC_TRNS,                                                             KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,              KC_TRNS,    KC_TRNS,    KC_TRNS),
+
+[3] = LAYOUT(
+  KC_TRNS,  KC_TRNS,  KC_TRNS,     KC_TRNS,    KC_TRNS,      KC_TRNS,       KC_TRNS,      KC_TRNS,      KC_TRNS,      KC_TRNS,    KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,      		 KC_TRNS,    KC_TRNS,    KC_TRNS,
+  KC_TRNS,  KC_TRNS,  KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,       KC_TRNS,      KC_TRNS,      KC_TRNS,      KC_TRNS,    KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,     KC_TRNS,    KC_TRNS,    KC_TRNS,
+  KC_TRNS,  KC_TRNS,  KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,       KC_TRNS,      KC_TRNS,      KC_TRNS,      KC_TRNS,    KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,       		 KC_TRNS ,   KC_TRNS,    KC_TRNS,
+  KC_TRNS,  KC_TRNS,  KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,       KC_TRNS,      KC_TRNS,      KC_TRNS,      KC_TRNS,    KC_TRNS,   KC_TRNS, KC_TRNS,  
+  KC_TRNS,  KC_TRNS,  KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,       KC_TRNS,      KC_TRNS,      KC_TRNS,      KC_TRNS,    KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,            		         KC_TRNS,
+  KC_TRNS,  KC_TRNS,  KC_TRNS,                               KC_TRNS,                                                             KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,              KC_TRNS,    KC_TRNS,    KC_TRNS)
```
```suggestion
  KC_TRNS,  KC_TRNS,  KC_TRNS,                               KC_TRNS,                                                             KC_TRNS,   KC_TRNS, KC_TRNS, KC_TRNS,              KC_TRNS,    KC_TRNS,    KC_TRNS)
```

zvecr commented on keyboards/candykeys/candy87/info.json:
```diff
@@ -0,0 +1,145 @@
+            "pixel_rain": true,
+            "rainbow_moving_chevron": true,
+            "raindrops": true,
+            "solid": true,
+            "solid_color": true,
```
```suggestion
```

rarepotato8de committed:
Update keyboards/candykeys/candy87/info.json

Co-authored-by: Joel Challis <git@zvecr.com>

rarepotato8de committed:
Update keyboards/candykeys/candy87/keymaps/default/keymap.c

Co-authored-by: Joel Challis <git@zvecr.com>

rarepotato8de committed:
Commit changes

Removed old rules.mk
info.json renamed to keyboard.json

rarepotato8de committed:
Added many layout options

iso, ansi, f12 and f13 layouts + community layouts

rarepotato8de committed:
All layouts

Added keymaps for all possible layouts

dunk2k commented on keyboards/candykeys/candy87/keyboard.json:
```diff
@@ -0,0 +1,892 @@
+        "driver": "vendor",
+        "pin": "GP7"
+    },
+	"community_layouts": ["tkl_f13_ansi_split_bs_rshift", "tkl_f13_ansi_tsangan_split_bs_rshift", "tkl_f13_iso_split_bs_rshift", "tkl_f13_iso_tsangan_split_bs_rshift", "tkl_ansi_split_bs_rshift", "tkl_ansi_tsangan_split_bs_rshift", "tkl_iso_split_bs_rshift", "tkl_iso_tsangan_split_bs_rshift"],
```
Could, remaining, compatible Community Layouts be added:
- tkl_ansi
- tkl_ansi_tsangan
- tkl_ansi_wkl
- tkl_ansi_wkl_split_bs_rshift
- tkl_iso
- tkl_iso_tsangan
- tkl_iso_wkl
- tkl_iso_wkl_split_bs_rshift
- tkl_f13_ansi
- tkl_f13_ansi_tsangan
- tkl_f13_ansi_wkl
- tkl_f13_ansi_wkl_split_bs_rshift
- tkl_f13_iso
- tkl_f13_iso_tsangan
- tkl_f13_iso_wkl
- tkl_f13_iso_wkl_split_bs_rshift

rarepotato8de replied:
just as option in the "community_layouts" array/list or do I also have to include the corresponding matrix layout definitions? Adding all of these layouts is a little overloaded, no?
Since WK Key will just be "ignored" if WKL is used. Same for "Fn" Key at the split right shift. Thanks for reply!

rarepotato8de committed:
All community layouts added

Added WKL and full backspace/ right shift community layouts as requested in my PR

waffle87 commented on keyboards/candykeys/candy87/candy87.c:
```diff
@@ -0,0 +1,34 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
```
```suggestion
#include "quantum.h"
```

waffle87 commented on keyboards/candykeys/candy87/keyboard.json:
Please run `qmk format-json -i` on this file.

rarepotato8de replied:
done. Also all other requests

waffle87 reviewed (commented):
A couple requests for the keymaps:
- RGB Matrix keycodes should be added to layer 1
- `QK_BOOT` keycode would also be nice to have on layer 1
- Replace keycodes with their short alias ie. `KC_PRINT_SCREEN` -> `KC_PSCR`, `KC_BACKSPACE` -> `KC_BSPC`, etc. Refer to the keycode list in the documentation.

rarepotato8de committed:
Update keyboards/candykeys/candy87/candy87.c

Co-authored-by: jack <jack@pngu.org>

rarepotato8de committed:
Keymap Update

Added RGB Matrix Keycodes, Aliases and formatting as requested for the PR

rarepotato8de commented:
Done a lot of changes

rarepotato8de committed:
Manufacturer Naming

Capital "K" for the brand

rarepotato8de commented:
it's still in progress/ waiting for approval
